src/main/java/net/minecraftforge/common/util/Constants.java

```diff
@@ -49,4 +51,168 @@
+         */
+        public static final int LINGERING_POTION_EFFECT         = 2007;
+        public static final int GATEWAY_SPAWN_EFFECTS           = 3000;
+        public static final int ENDERMAN_GROWL_SOUND            = 3001;
+    }
+
+
+    /**
+     * The flags used when calling {@link net.minecraft.world.World#setBlockState}<br>
+     * Can be found from {@link net.minecraft.world.World#markAndNotifyBlock} and {@link net.minecraft.client.renderer.RenderGlobal#notifyBlockUpdate}<br>
+     * Flags can be combined with bitwise OR
+     */
+    public static class BlockFlags {
+        /**
+         * Calls neighborChanged on surrounding blocks
+         */
+        public static final int NOTIFY_NEIGHBORS     = 0b00001;
+        /**
+         * Sends the update to the client
+         */
+        public static final int SEND_TO_CLIENTS      = 0b00010;
+        /**
+         * Stops the blocks from being marked for a render update
+         */
+        public static final int NO_RERENDER          = 0b00100;
+        /**
+         * Makes the block be re-rendered immediately, on the main thread.
+         * If NO_RERENDER is set, then this will be ignored
+         */
+        public static final int RERENDER_MAIN_THREAD = 0b01000;
+        /**
+         * Disables observers from seeing this update
+         */
+        public static final int NO_OBSERVERS         = 0b10000;
+
+        public static final int DEFAULT = NOTIFY_NEIGHBORS | SEND_TO_CLIENTS;
+        public static final int DEFAULT_AND_RERENDER = DEFAULT | RERENDER_MAIN_THREAD;
+    }
+
+    /**
+     * The mutex bits used for AI tasks {@link net.minecraft.entity.ai.EntityAIBase#setMutexBits(int)}
+     * Bits can be combined with bitwise OR
+     */
+    public static class AiMutexBits {
+        public static final int MOVE = 0b001;
+        public static final int LOOK = 0b010;
+        public static final int JUMP = 0b100;
+    }
+
+    /**
+     * The flags used for {@link net.minecraft.entity.Entity#getFlag(int)} and {@link net.minecraft.entity.Entity#setFlag(int, boolean)}}<br>
+     * Can be found by searching for the usages of getFlag
+     */
+    public static class EntityFlags {
+        public static final int BURNING       = 0;
+        public static final int SNEAKING      = 1;
+        public static final int SPRINTING     = 3;
+        public static final int INVISIBLE     = 5;
+        public static final int GLOWING       = 6;
+        public static final int ELYTRA_FLYING = 7;
+    }
+
+    /**
+     * The operation used for Attribute modifier operations, {@link net.minecraft.entity.ai.attributes.AttributeModifier#AttributeModifier(String, double, int)}<br>
+     * Can be found at {@link ModifiableAttributeInstance#computeValue()}<br>
+     * Order of operations are {@link #ADD}, {@link #ADD_MULTIPLE}, {@link #MULTIPLY}
+     */
+    public static class AttributeModifierOperation {
+        /**
+         * The modifier value is added onto the total value
```

Comment:
Should say added on to the base value, rather than total

Reply:
its not added onto the base value, its added to total value. However the total value starts with the base value, which is somthing I should probably say

Reply:
Maybe it should say "added to the base value to produce the value used in other operations". I think "added to the total value" could mislead people to think that it is not affected by the other operations.

Reply:
I could state on the class descriptor that the total value starts with the base value

Reply:
The minecraft wiki explains it really well. Maybe worth linking to that on the class?
https://minecraft.gamepedia.com/Attribute#Operations